src/processor/operator/persistent/reader/csv/parallel_csv_reader.cpp

```diff
@@ -215,24 +226,19 @@ static std::unique_ptr<TableFuncSharedState> initSharedState(TableFunctionInitIn
 
     for (idx_t i = 0; i < sharedState->readerConfig.getNumFiles(); ++i) {
         auto filePath = sharedState->readerConfig.filePaths[i];
-        auto reader = std::make_unique<ParallelCSVReader>(filePath, csvOption.copy(),
+        auto reader = std::make_unique<ParallelCSVReader>(filePath, i, csvOption.copy(),
             columnInfo.copy(), bindData->context, nullptr);
         sharedState->totalSize += reader->getFileSize();
     }
+
     return sharedState;
 }
 
 static std::unique_ptr<TableFuncLocalState> initLocalState(TableFunctionInitInput& /*input*/,
-    TableFuncSharedState* state, storage::MemoryManager* /*mm*/) {
+    TableFuncSharedState* /*state*/, storage::MemoryManager* /*mm*/) {
     auto localState = std::make_unique<ParallelCSVLocalState>();
-    auto sharedState = state->ptrCast<ParallelCSVScanSharedState>();
-    localState->fileIdx = 0;
-    localState->errorHandler = std::make_unique<LocalCSVFileErrorHandler>(
-        sharedState->context->getClientConfig()->warningLimit, sharedState->csvOption.ignoreErrors,
-        &sharedState->errorHandlers[localState->fileIdx]);
-    localState->reader = std::make_unique<ParallelCSVReader>(sharedState->readerConfig.filePaths[0],
-        sharedState->csvOption.copy(), sharedState->columnInfo.copy(), sharedState->context,
-        localState->errorHandler.get());
```

Comment:
This was removed as the error handler and reader are already constructed in `tableFunc` and this prevents `reportFinishedBlock` from being called twice for `blockIdx == 0`